compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -144,7 +144,17 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
     /** Is a reference to a class but not `this.type` */
     def isClassRef = sym.isClass && !tp.isInstanceOf[ThisType]
 
-    if (!sym.exists || levelOK(sym))
+    /** Is this a static path or a type porjection with a static prefix */
+    def isStaticPathOK(tp1: Type): Boolean = tp1 match {
+      case tp1: TypeRef => tp1.symbol.is(Package) || isStaticPathOK(tp1.prefix)
+      case tp1: TermRef => tp1.symbol.isStaticOwner || isStaticPathOK(tp1.prefix)
+      case tp1: ThisType => tp1.tref.symbol.is(Package) || tp1.tref.symbol.is(Module)
```

Comment:
Typo: "porjection"

I don't think this is the right logic for isStaticPathOK. How about:
```
tp1.stripTypeVar match
  case tp1: TypeRef => tp1.symbol.is(Package) || isStaticPathOK(tp1.prefix)
  case tp1: TermRef => tp1.symbol.is(Module) && tp1.symbol.isStatic
  case tp1: ThisType => tp1.cls.isStaticOwner
  case tp1: AppliedType => isStaticPathOK(tp1.tycon)

Reply:
I used this version it but made a couple of extra changes to make all the tests pass